consensus/istanbul/backend/engine.go

```diff
@@ -786,20 +877,34 @@ func assembleExtra(header *types.Header, oldValSet []istanbul.ValidatorData, new
 			"addedValidators", common.ConvertToStringSlice(addedValidatorsAddresses), "removedValidators", removedValidators.Text(16))
 	}
 
-	ist := &types.IstanbulExtra{
+	// if the extras could not be calculated, initialize a new struct with
```

Comment:
Should we be trying to `ExtractIstanbulExtra` before this we create this struct, much like we do for `writeCommittedSeals`?

Reply:
I think I like it as is, do you mind me not changing it? Don't think the attempt to create consistency between the two is that valuable here